cmd/tetra/getevents/getevents.go

```diff
@@ -151,6 +161,19 @@ func New() *cobra.Command {
 				return fmt.Errorf("invalid value for %q flag: %s", "color", Options.Color)
 			}
 
+			var supportedEventTypes string
+
+			for _, v := range tetragon.EventType_name {
+				supportedEventTypes += v + ", "
+			}
+			supportedEventTypes = strings.TrimSuffix(supportedEventTypes, ", ")
+
+			for _, v := range Options.EventTypes {
+				if _, ok := tetragon.EventType_value[v]; !ok {
+					return fmt.Errorf("invalid value for %q flag: %s. Supported are %s", "event-types", v, supportedEventTypes)
```

Comment:
you do return an error here (that we retrieve in the test output), which will make the command error, I would suggest just filtering dynamically with the provided eventtypes without checking if it exists. filtering by DOESNOTEXIST should just return nothing as you wrote in your test?

Reply:
The idea is that when a user passes an argument that is not any of:
```
PROCESS_KPROBE, PROCESS_LOADER, TEST, PROCESS_UPROBE, RATE_LIMIT_INFO, UNDEF, PROCESS_EXEC, PROCESS_EXIT, PROCESS_TRACEPOINT
```

it should fail. 

Now, the following test always returns `0`

```
	t.Run("FilterInexistent", func(t *testing.T) {
		testutils.MockPipedFile(t, testutils.RepoRootPath("testdata/events.json"))
		cmd := New()
		cmd.SetArgs([]string{"--event-types", "INEXISTENT"})
		output := testutils.RedirectStdoutExecuteCmd(t, cmd)
		assert.Equal(t, 0, bytes.Count(output, []byte("\n")))
	})
```

How can I then sometimes fail and sometimes pass? 

Two attempts:
![image](https://github.com/cilium/tetragon/assets/9934402/3ea284da-7ce5-4720-a4ac-290b5f2a2e45)

Reply:
I think it's `RedirectStdoutExecuteCmd` that doesn't expect the command to fail (return something different than 0) and maybe there's a race somewhere in the piping explaining the broken pipe thingie. I admit I cannot reproduce on my machine even running dozens of times without caching :/

Reply:
Interesting. I wonder if we really need a test for the failing case or if we can just remove it.

Reply:
let me propose you something

Reply:
ah we are seeing the panic because it's `MockPipedFile` that doesn't support the cmd to fail, stdin is still piped to the entry of the command which exits before the end of the test because of the error.